README.md

```diff
@@ -1 +1,81 @@
-# Android Error Handlesr
\ No newline at end of file
+## Structure
+  The library consist of 2 modules:
+  - **error-handler-core** - defines the DataLayerException, the root exception for an entire data layer, and ResultWrapper, wrapper class based on Railway Oriented Programming pattern.
+  - **error-handler-rest** - which provides error handling for REST API consumption. Note that this module is built on top of [retrofit-error-wrapping](https://github.com/halcyonmobile/retrofit-error-wrapping) library. This module provides subclasses for DataLayerException with possible failures that can occur during an HTTP call.
+
+## Setup
+  ###### Latest stable version is `0.1.0`
+  - Add `jcenter` to your project `build.gradle`
+   ```kotlin 
+repositories {
+       // other repositories
+       jcenter()
+}
+   ``` 
+   - Add this library to your project
+   ```kotlin
+dependencies {
+       // other dependencies
+       implementation "com.halcyonmobile.error-handler:core:<latest_version>"
```

Comment:
We should add something similar if the rest will also include the core:
`// if you want to use only the core module
implementation "com.halcyonmobile.error-handler:core:<latest_version>"
// if you want the core and rest modules as well
implementation "com.halcyonmobile.error-handler:rest:<latest_version>"`